core/corehttp/gateway_handler.go

```diff
@@ -75,39 +68,7 @@ func (i *gatewayHandler) loadTemplate() error {
 	return nil
 }
 
-func (i *gatewayHandler) resolveNamePath(ctx context.Context, p string) (string, error) {
-	p = gopath.Clean(p)
-
-	if strings.HasPrefix(p, IpnsPathPrefix) {
-		elements := strings.Split(p[len(IpnsPathPrefix):], "/")
-		hash := elements[0]
-		rp, err := i.node.Namesys.Resolve(ctx, hash)
-		if err != nil {
-			return "", err
-		}
-
-		elements = append(rp.Segments(), elements[1:]...)
-		p = gopath.Join(elements...)
-	}
-	if !strings.HasPrefix(p, IpfsPathPrefix) {
-		p = gopath.Join(IpfsPathPrefix, p)
-	}
-	return p, nil
-}
-
-func (i *gatewayHandler) ResolvePath(ctx context.Context, p string) (*dag.Node, string, error) {
-	p, err := i.resolveNamePath(ctx, p)
-	if err != nil {
-		return nil, "", err
-	}
-
-	node, err := i.node.Resolver.ResolvePath(path.Path(p))
-	if err != nil {
-		return nil, "", err
-	}
-	return node, p, err
-}
-
+// TODO(cryptix):  these four helper funcs shoudl also be available elsewhere, i think?
 func (i *gatewayHandler) NewDagFromReader(r io.Reader) (*dag.Node, error) {
```

Comment:
There's the stuff in `importer` that this method is using.  I guess this is just for the convenience of binding it to the `gatewayHandler`?